connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -636,6 +642,180 @@ public void run() {
         PowerMock.verifyAll();
     }
 
+    // Verify that when commitAsync is called but the supplied callback is not called by the consumer before a
+    // rebalance occurs, the async callback does not reset the last committed offset from the rebalance.
+    // See KAFKA-5731 for more information.
+    @Test
+    public void testCommitWithOutOfOrderCallback() throws Exception {
```

Comment:
This test case is a bit of a pain to read, so I can only imagine how much fun it was to write 😉 . At a glance, I'm wondering if it would have been simpler to capture the commit callbacks so that it's easier to invoke them in the order you want. Thinking longer term, it might be nice to consider refactoring this code to make testing simpler.